### ISSUE TITLE
Saving Preset of Songs

### ISSUE THREAD

**Issue by fartleako** [reckel-jm/cantara#7]
It would be good if you can save a list of songs beforehand (in a file for example) and just open this file when needed. After this, all songs should be loaded and ready to display.

These files should be created through Cantara itself.

**Comment by reckel-jm** [comment]
Thanks for bringing up this feature request! A simple solution to export import lists of selected songs is already implemented and can be found under Menu File -> Import/Export song list. However, this creates only a text file with the path of the song files, not the content of them. This would be a nice to add feature! Steps which have to be done to implement this would be:

- [ ] Define a file format which can display multiple songs, its titles and its content. At the moment, I am thinking of a TeX-style pseudo-format which would look like this:

```
\begincantarasong{Amazing-Grace.song}
% filecontent here
\endcantarasong
\begincantarasong{How great Thou art.song}
% filecontent here
\endcantarasong
```

XML or JSON would be also possible, but I am afraid that some chars of the content of the song files would be needed to escape or there are other kind of incompatibilities do deal with.

- [ ] Deal with the import of new songs: What happens, if a user would like to import a song list with content, but he has already a file with the same name on his PC? What if the file is not the same, how to deal with different versions?

- [ ] Write a class to create and parse such kind of files.

**Comment by reckel-jm** [comment]
Commit 960d196bde40fe06e0de1e7e36e189f1e7297f82 implements an export function for a newly defined SongTeX-format (in honor of the great Computer Scientist and Musician Donald Knuth).

**Comment by reckel-jm** [comment]
The latest commits (e.g. db2094767f0cdc34d962f5905ad98d8d52671380) on branch master include a feature to import SongTeX files which consists of the order and the content of the song files.
The song files will be added to the repository. If they already exist and aren't equal, they would be added under a different name.
Please check and report bugs/etc. if necessary!

**Comment by reckel-jm** [comment]
Feature is finally implemented in v2.4.0beta (680f78d76539e8023fc697dafd116de0579626c5).
